control-plane/csi-driver/src/bin/node/node.rs

```diff
@@ -432,9 +438,149 @@ impl node_server::Node for Node {
+                            "xfs" => Self::SupportedFilesystem(FileSystem::from(Fs::Xfs)),
+                            "btrfs" => Self::SupportedFilesystem(FileSystem::from(Fs::Btrfs)),
+                            alien_fs => Self::UnsupportedFilesystem(format!(
+                                "'{alien_fs}' is not a supported filesystem"
+                            )),
+                        },
+                        AccessType::Block(_) => Self::NotAFilesystem,
+                    },
+                    Err(error) => {
+                        Self::Unknown(format!("couldn't determine CSI AccessType type: {}", error))
+                    }
+                }
+            }
+        }
+        //===============================CsiAccessType=============================================
+
+        let request = request.into_inner();
+
+        let vol_uuid = Uuid::parse_str(request.volume_id.as_str()).map_err(|error| {
+            failure!(
+                Code::InvalidArgument,
+                "Malformed volume UUID '{}': {}",
+                request.volume_id,
+                error
+            )
+        })?;
+
+        if request.volume_path.is_empty() {
+            return Err(failure!(Code::InvalidArgument, "'volume_path' is empty"));
+        }
+
+        let required_bytes = request
+            .capacity_range
+            .as_ref()
+            .ok_or(failure!(
+                Code::InvalidArgument,
+                "Cannot expand volume '{}': invalid request {:?}: missing CapacityRange",
+                request.volume_id,
+                request
+            ))?
+            .required_bytes;
+
+        let _guard = VolumeOpGuard::new(vol_uuid)?;
+
+        let dev_name = Device::lookup(&vol_uuid)
+            .await
+            .map_err(|error| {
+                failure!(
+                    Code::Internal,
+                    "device lookup error for volume '{}': {}",
+                    vol_uuid,
+                    error
+                )
+            })?
+            .ok_or(failure!(
+                Code::InvalidArgument,
+                "failed to find a device for volume {}",
+                vol_uuid
+            ))?
+            .devname();
+
+        // Get device size.
+        // The underlying block device should already have been expanded to the
+        // required size as a part of the ControllerExpandVolume call.
+        let capacity_bytes = get_size_from_dev_name(dev_name.as_str()).map_err(|error| {
+            failure!(
+                Code::Internal,
+                "failed to find the device size of device {}: {}",
+                dev_name,
+                error
+            )
+        })? as i64;
+
+        // Ensure capacity_bytes is greater than or equal ot required_bytes.
+        if capacity_bytes.lt(&required_bytes) {
+            return Err(failure!(
+                Code::FailedPrecondition,
+                "block device capacity is invalid"
```

Comment:
nit: something more like these lines
```suggestion
                "block device capacity is lower than the required"
```